### PR TITLE
New icqsol_add_texture tool

### PR DESCRIPTION
**Pull request by pletzer**
Greg, I added the icqsol_add_texture tool. --Alex

### CODE REVIEW

File: tools/icqsol_add_texture/icqsol_add_texture.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+import argparse
+import shutil
+from pkg_resources import resource_filename
```
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py:4]: @pletzer This complexity needs to be encapsulated in the Shape Manager - similar to not importing numpy into Galaxy tool wrappers, underlying components like resource_filename should not be imported here either.

File: tools/icqsol_add_texture/icqsol_add_texture.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Add texture. We use setuptools, which installs the package as a python "egg", 
+# essentially a zip file containing the modules and data files. Must use resource_filename
+# to access the data files.
+texture_name_to_filename = {
+    'stone': resource_filename('icqsol', 'textures/220px-COnglomerate-sandstone_layers_Nerriga.jpg'),
+    'wood': resource_filename('icqsol', 'textures/Swietenia_macrophylla_wood.jpg'),
+    'checkerboard': resource_filename('icqsol', 'textures/checkerboard.png'),
+}
+
```
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py:41]: We should not have hard-coded file names here - these should be command line options.

File: tools/icqsol_add_texture/icqsol_add_texture.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Selected texture is "stone" if the texture name does not match any of our values.s
+texture_file = texture_name_to_filename.get(args.texture, 'stone')
```
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py]: The above should be handled by a method in the ShapeManager.

File: tools/icqsol_add_texture/icqsol_add_texture.xml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            </actions>
+        </data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" value="box_with_surface_field.vtkascii" ftype="vtkascii" />
+            <param name="input_file_format_and_type" value="vtkascii" />
+            <param name="input_dataset_type" value="POLYDATA" />
+            <param name="max_edge_length" value="0.1" />
+            <output name="output" file="box_refined.vtkascii" ftype="vtkascii" />
```
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.xml:49]: This should be a dataset the is the result of adding a texture to a shape.

Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.xml:49]: @pletzer I'll merge this and perform the fixes.  Thanks!

File: tools/icqsol_add_texture/icqsol_add_texture.py (hunk 2 above)
Comment (pletzer) [inline tools/icqsol_add_texture/icqsol_add_texture.py:41]: @gregvonkuster We could but then the user would need to upload png or jpeg files. (ShapeManager's addTexture method takes a file name).

File: tools/icqsol_add_texture/icqsol_add_texture.py (hunk 1 above)
Comment (pletzer) [inline tools/icqsol_add_texture/icqsol_add_texture.py:4]: Greg,
This is the only and recommended way to access data files. We're already assuming setuptools installed and pkg_resources comes with it. There is no additional installation. 

--Alex

File: tools/icqsol_add_texture/icqsol_add_texture.py (hunk 2 above)
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py:41]: @pletzer The texture file is passed to the ShapeManager, just like Shape itself, right?  So bot start out as file paths, and the ShapeManager just loads the shape before this particular method is call.  See my fixes that I made to the icqsol_add_texture tool.

File: tools/icqsol_add_texture/icqsol_add_texture.py (hunk 1 above)
Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py:4]: @pletzer with Galaxy tools, the data file is uploaded to a history and selected by the Galaxy tool from the history and passed as a file path to the underlying package code.

File: tools/icqsol_add_texture/icqsol_add_texture.py (hunk 2 above)
Comment (pletzer) [inline tools/icqsol_add_texture/icqsol_add_texture.py:41]: @gregvonkuster 

I tried to merge and the code is now broken. Not sure what you're trying to achieve. The line "from pkg_resource import resource_file" was removed and this breaks 

# Add texture. We use setuptools, which installs the package as a python "egg",

# essentially a zip file containing the modules and data files. Must use resource_filename

# to access the data files.

texture_name_to_filename = {
    'stone': resource_filename('icqsol', 'textures/220px-COnglomerate-sandstone_layers_Nerriga.jpg'),
    'wood': resource_filename('icqsol', 'textures/Swietenia_macrophylla_wood.jpg'),
    'checkerboard': resource_filename('icqsol', 'textures/checkerboard.png'),
}

If the users are expected to provide the texture files then the above needs to be removed. 

--Alex

Comment (gregvonkuster) [inline tools/icqsol_add_texture/icqsol_add_texture.py:41]: @pletzer It is removed in my repo.  Try a fresh clone from my repo.  I don't know what is going on with your repo.  Make sure not to commit anything to my repo until you have a clean one to work with.

**Pull request merged**